pkg/liqonet/tunnel/wireguard/driver.go

```diff
@@ -505,18 +549,30 @@ func (w *Wireguard) updateConnectionStatus(oldConn *netv1alpha1.Connection) (*ne
 	}
 
 	for i := range wgDev.Peers {
-		if wgDev.Peers[i].PublicKey.String() == oldConn.PeerConfiguration[liqoconst.PublicKey] {
-			// Peer has been found.
-			peer := wgDev.Peers[i]
-			// Check if the handshake with the remote peer completed.
-			if !peer.LastHandshakeTime.IsZero() {
-				oldConn.Status = netv1alpha1.Connected
-				oldConn.StatusMessage = netv1alpha1.ConnectedMessage
-				return oldConn, nil
```

Comment:
This function should no longer be needed, as the check whether the tunnel is established is performed based on the connection checker, and not the last handshake (which should also be disabled).